homeassistant/components/group/__init__.py

```diff
@@ -222,10 +223,12 @@ def groups_with_entity(hass: HomeAssistant, entity_id: str) -> list[str]:
     return groups
 
 
-async def async_setup(hass, config):
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up all groups found defined in the configuration."""
-    if (component := hass.data.get(DOMAIN)) is None:
-        component = hass.data[DOMAIN] = EntityComponent(_LOGGER, DOMAIN, hass)
+    if DOMAIN not in hass.data:
+        hass.data[DOMAIN] = EntityComponent(_LOGGER, DOMAIN, hass)
+
+    component = hass.data[DOMAIN]
```

Comment:
Was just thinking, this makes it `Any`, which is suboptimal as well.

This works better:

```suggestion
    component: EntityComponent = hass.data[DOMAIN]
```

It does reveal new issues though.

Reply:
🤷  depends on the changeset.

I'm fine with 1 or 3, 2 is a bit tricky actually as everything seems fine but isn't.

Reply:
I have reverted the changes to `group`.
It looks like it will need quite a bit of work to get working.

Reply:
(and it  shouldmake CI run partial instead of full)

Reply:
It does :)
Pretty darn happy with that CI stuff nowadays 👯